test/metabase/db/schema_migrations_test.clj

```diff
@@ -1554,7 +1562,35 @@
                                                       (assoc :cache_ttl 10)))
       (migrate!)
       (is (= []
-             (t2/select :model/CacheConfig))))))
+             (t2/select :cache_config))))))
+
+(deftest cache-config-mysql-update-test
+  (when (= (mdb/db-type) :mysql)
```

Comment:
This doesn't stop the test from running on H2 when the application database is MySQL.

If you execute 
`(mt/set-test-drivers! [:h2])`
And then run this test, it will run the test migrations on an H2 database and the test will fail because the MySQL migration doesn't run.

You need to change this to use mysql for the tests
```suggestion
  (mt/test-driver :mysql
```